### PR TITLE
Run publish-packages in parallel

### PR DESCRIPTION
**Pull request by andy-ms**

### CODE REVIEW

File: src/lib/package-publisher.ts [hunk 1]
```diff
@@ -1,79 +1,94 @@
...
+				const stderr = <string> <any> stderrBuffer;
+				if (err) {
+					log.error(`${commandDescription} failed: ${JSON.stringify(err)}`);
+					log.info(`${commandDescription} failed, refer to error log`);
+					log.error(<string> <any> stderr);
+					resolve(false);
+				}
+				else {
+					log.info("Ran successfully");
+					log.info(<string> <any> stdout);
```
Comment (RyanCavanaugh) [inline src/lib/package-publisher.ts]: `stdout` already `string`

File: src/lib/package-publisher.ts [hunk 2]
```diff
@@ -1,79 +1,94 @@
...
-		}
+		return new Promise((resolve, reject) => {
+			child_process.exec(cmd, { encoding: "utf8" }, (err, stdoutBuffer, stderrBuffer) => {
+				// These are wrongly typed as Buffer.
+				const stdout = <string> <any> stdoutBuffer;
+				const stderr = <string> <any> stderrBuffer;
+				if (err) {
+					log.error(`${commandDescription} failed: ${JSON.stringify(err)}`);
+					log.info(`${commandDescription} failed, refer to error log`);
+					log.error(<string> <any> stderr);
```
Comment (RyanCavanaugh) [inline src/lib/package-publisher.ts]: `stderr` already `string`

File: src/publish-packages.ts [hunk 3]
```diff
@@ -2,45 +2,50 @@ import * as fs from "fs";
...
-		publishLog.errors.forEach(err => {
-			log.push(`   * ERROR: ${err}`);
-			console.log(` Error! ${err}`);
+	if (unpublish) {
+		for (const pkg of allPackages) {
+			await publisher.unpublishPackage(pkg, dry);
+		}
+	}
+	else {
+		await nAtATime(100, allPackages, async typing => {
```
Comment (RyanCavanaugh) [inline src/publish-packages.ts]: I'd be somewhat surprised if 100 works, but :+1: if it does

**Comment by RyanCavanaugh** [comment]
:+1: with comments

**Comment by andy-ms** [comment]
@RyanCavanaugh, second commit OK? https://github.com/Microsoft/types-publisher/pull/30/commits/07b122138f584748c3b09e9c843fdfd5b1e1a409

**Comment by RyanCavanaugh** [comment]
Yep

**Pull request merged**